src/test/Fuzzers.sol

```diff
@@ -10,35 +10,97 @@ import {BalanceDelta} from "../types/BalanceDelta.sol";
+import {SafeCast} from "../../src/libraries/SafeCast.sol";
 
 contract Fuzzers is StdUtils {
+    using SafeCast for uint256;
+
     Vm internal constant _vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
 
-    function boundLiquidityDelta(PoolKey memory key, int256 liquidityDelta) internal pure returns (int256) {
-        return bound(
-            liquidityDelta, 0.0000001e18, int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)) / 2)
+    function boundLiquidityDelta(PoolKey memory key, int256 liquidityDeltaUnbounded, int256 liquidityMaxByAmount)
+        internal
+        pure
+        returns (int256)
+    {
+        int256 liquidityMaxPerTick = int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)));
+
+        // Finally bound the seeded liquidity by either the max per tick, or by the amount allowed in the position range.
+        int256 liquidityMax = liquidityMaxByAmount > liquidityMaxPerTick ? liquidityMaxPerTick : liquidityMaxByAmount;
+        return bound(liquidityDeltaUnbounded, 0.0000001e18, liquidityMax);
+    }
+
+    // Uses tickSpacingToMaxLiquidityPerTick/2 as one of the possible bounds.
+    // Potentially adjust this value to be more strict for positions that touch the same tick.
+    function boundLiquidityDeltaStrict(PoolKey memory key, int256 liquidityDeltaUnbounded, int256 liquidityMaxByAmount)
+        internal
+        pure
+        returns (int256)
+    {
+        // Divide by half to bound liquidity more. TODO: Probably a better way to do this.
+        int256 liquidityMaxTightBound = int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)) / 2);
+
+        // Finally bound the seeded liquidity by either the max per tick, or by the amount allowed in the position range.
+        int256 liquidityMax =
+            liquidityMaxByAmount > liquidityMaxTightBound ? liquidityMaxTightBound : liquidityMaxByAmount;
+        return bound(liquidityDeltaUnbounded, 1, liquidityMax);
+    }
+
+    function getLiquidityDeltaFromAmounts(int24 tickLower, int24 tickUpper, uint160 sqrtPriceX96)
+        internal
+        pure
+        returns (int256)
+    {
+        // First get the maximum amount0 and maximum amount1 that can be deposited at this range.
+        (uint256 maxAmount0, uint256 maxAmount1) = LiquidityAmounts.getAmountsForLiquidity(
+            sqrtPriceX96,
+            TickMath.getSqrtPriceAtTick(tickLower),
+            TickMath.getSqrtPriceAtTick(tickUpper),
+            uint128(type(int128).max)
         );
+
+        // Compare the max amounts (defined by the range of the position) to the max amount constrained by the type container.
+        // The true maximum should be the minimum of the two.
+        // (ie If the position range allows a deposit of more then int128.max in any token, then here we cap it at int128.max.)
+
+        uint256 amount0 = uint256(type(uint128).max / 2);
```

Comment:
why is this being divided by 2?

Reply:
why not just do type(int128).max?

Reply:
type(uint128).max / 2 == type(int128).max